<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="1" ignored="0" total="5" passed="4">
  <reporter-output>
  </reporter-output>
  <suite name="Example Suite for my tests" duration-ms="958" started-at="2022-01-04T11:36:48 EST" finished-at="2022-01-04T11:36:49 EST">
    <groups>
    </groups>
    <test name="Injection Tests" duration-ms="77" started-at="2022-01-04T11:36:48 EST" finished-at="2022-01-04T11:36:49 EST">
      <class name="com.revature.tests.injection.MathsTests">
        <test-method status="PASS" signature="checkDivision()[pri:0, instance:com.revature.tests.injection.MathsTests@62ee68d8]" name="checkDivision" duration-ms="8" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- checkDivision -->
      </class> <!-- com.revature.tests.injection.MathsTests -->
    </test> <!-- Injection Tests -->
    <test name="Holder Tests" duration-ms="843" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
      <class name="com.revature.tests.holder.MockingAndStubbing">
        <test-method status="PASS" signature="setup()[pri:0, instance:com.revature.tests.holder.MockingAndStubbing@31ef45e3]" name="setup" is-config="true" duration-ms="769" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="checkMethodCallAndInputs()[pri:0, instance:com.revature.tests.holder.MockingAndStubbing@31ef45e3]" name="checkMethodCallAndInputs" duration-ms="49" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <exception class="org.mockito.exceptions.verification.ArgumentsAreDifferent">
            <message>
              <![CDATA[Argument(s) are different! Wanted:
maths.division(20, 4);
-> at com.revature.tests.holder.MockingAndStubbing.checkMethodCallAndInputs(MockingAndStubbing.java:45)
Actual invocations have different arguments:
maths.division(20, 5);
-> at dev.revature.holder.Calculator.evenOdd(Calculator.java:14)
]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.mockito.exceptions.verification.ArgumentsAreDifferent: Argument(s) are different! Wanted:
maths.division(20, 4);
-> at com.revature.tests.holder.MockingAndStubbing.checkMethodCallAndInputs(MockingAndStubbing.java:45)
Actual invocations have different arguments:
maths.division(20, 5);
-> at dev.revature.holder.Calculator.evenOdd(Calculator.java:14)

at com.revature.tests.holder.MockingAndStubbing.checkMethodCallAndInputs(MockingAndStubbing.java:45)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.util.ArrayList.forEach(ArrayList.java:1257)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
]]>
            </full-stacktrace>
          </exception> <!-- org.mockito.exceptions.verification.ArgumentsAreDifferent -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- checkMethodCallAndInputs -->
        <test-method status="PASS" signature="checkNumberOfMethodCalls()[pri:0, instance:com.revature.tests.holder.MockingAndStubbing@31ef45e3]" name="checkNumberOfMethodCalls" duration-ms="3" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- checkNumberOfMethodCalls -->
        <test-method status="PASS" signature="controlReturnValueWithStubbing()[pri:0, instance:com.revature.tests.holder.MockingAndStubbing@31ef45e3]" name="controlReturnValueWithStubbing" duration-ms="8" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- controlReturnValueWithStubbing -->
        <test-method status="PASS" signature="determineReturnValue()[pri:0, instance:com.revature.tests.holder.MockingAndStubbing@31ef45e3]" name="determineReturnValue" duration-ms="1" started-at="2022-01-04T11:36:49 EST" finished-at="2022-01-04T11:36:49 EST">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- determineReturnValue -->
      </class> <!-- com.revature.tests.holder.MockingAndStubbing -->
    </test> <!-- Holder Tests -->
  </suite> <!-- Example Suite for my tests -->
</testng-results>
